Fix custom-device default forwardPortSuccessRegex

mbriand opened the pull request:
This PR tries to make port forwarding verification on custom devices more reliable, by not relying on behavior that may change across different systems.

Previously we were looking for `Linux` word in target connection banner. Now we ask the remote to print a fixed string ('Port forwarding success`) and look for this given string to ensure the connection was successful.

Fix #97641 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

mbriand committed:
Fix custom-device default forwardPortSuccessRegex

christopherfujino reviewed (commented):
I think this would need a test before we can merge.

mbriand committed:
Fix custom-devices tests for forwardPort

mbriand committed:
Fix custom-device postBuildCommand in JSON

Having an empty list as postBuildCommand value will trigger an error on
JSON parsing: make sure to save a null instead.

mbriand committed:
Add back deleted flutter devices tests

Tests were removed in #98299

mbriand commented:
Sorry, I was not very active on this PR.

So yes, I would be happy to add some tests here. Actually my modification made some tests to fail, but these tests were removed in commit 87fe3fe03149 (#98299), as they were failing for another issue, that I believe is actually a bug.

I managed to fix this bug and make the test pass, would you mind if I import back these tests ? I'm pushing these modifications, just to detail what I am talking about.

christopherfujino commented:
> Sorry, I was not very active on this PR.
> 
> So yes, I would be happy to add some tests here. Actually my modification made some tests to fail, but these tests were removed in commit [87fe3fe](https://github.com/flutter/flutter/commit/87fe3fe03149fc14b70c4c3daf37e9a4b17d49fe) (#98299), as they were failing for another issue, that I believe is actually a bug.
> 
> I managed to fix this bug and make the test pass, would you mind if I import back these tests ? I'm pushing these modifications, just to detail what I am talking about.

Oh nice! This is great.

christopherfujino commented:
The analysis failure was fixed upstream, a rebase should fix

christopherfujino reviewed (approved):
LGTM

mbriand committed:
fixup! Fix custom-device postBuildCommand in JSON

mbriand commented:
Sorry, I still have an issue with the "Linux analyze" test. It complains that I have to "Avoid redundant argument values", but if I remove the `postBuildCommand` parameter, it complains that "The parameter 'postBuildCommand' is required".

```
   info â�¢ Avoid redundant argument values â�¢ packages/flutter_tools/test/commands.shard/hermetic/custom_devices_test.dart:557:33 â�¢ avoid_redundant_argument_values
   info â�¢ Avoid redundant argument values â�¢ packages/flutter_tools/test/commands.shard/hermetic/custom_devices_test.dart:647:33 â�¢ avoid_redundant_argument_values
   info â�¢ Avoid redundant argument values â�¢ packages/flutter_tools/test/commands.shard/hermetic/custom_devices_test.dart:737:33 â�¢ avoid_redundant_argument_values
   info â�¢ Avoid redundant argument values â�¢ packages/flutter_tools/test/commands.shard/hermetic/custom_devices_test.dart:831:33 â�¢ avoid_redundant_argument_values
   info â�¢ Avoid redundant argument values â�¢ packages/flutter_tools/test/commands.shard/hermetic/custom_devices_test.dart:911:33 â�¢ avoid_redundant_argument_values
   info â�¢ Avoid redundant argument values â�¢ packages/flutter_tools/test/commands.shard/hermetic/custom_devices_test.dart:1236:33 â�¢ avoid_redundant_argument_values
```

```
warning â�¢ The parameter 'postBuildCommand' is required â�¢ packages/flutter_tools/test/commands.shard/hermetic/custom_devices_test.dart:546:13 â�¢ missing_required_param
warning â�¢ The parameter 'postBuildCommand' is required â�¢ packages/flutter_tools/test/commands.shard/hermetic/custom_devices_test.dart:635:13 â�¢ missing_required_param
warning â�¢ The parameter 'postBuildCommand' is required â�¢ packages/flutter_tools/test/commands.shard/hermetic/custom_devices_test.dart:723:13 â�¢ missing_required_param
warning â�¢ The parameter 'postBuildCommand' is required â�¢ packages/flutter_tools/test/commands.shard/hermetic/custom_devices_test.dart:817:19 â�¢ missing_required_param
warning â�¢ The parameter 'postBuildCommand' is required â�¢ packages/flutter_tools/test/commands.shard/hermetic/custom_devices_test.dart:896:13 â�¢ missing_required_param
warning â�¢ The parameter 'postBuildCommand' is required â�¢ packages/flutter_tools/test/commands.shard/hermetic/custom_devices_test.dart:1219:13 â�¢ missing_required_param
```

So I explicitly added `// ignore: avoid_redundant_argument_values`. Is that the right way to do it ?

christopherfujino commented:
@Jasguerrero can you review this one too?

Jasguerrero reviewed (approved):
LGTM
